objects/obj_pnunit/Alarm_5.gml

```diff
@@ -1,95 +1,96 @@
-total_battle_exp_gain = 0;
-if (obj_ncombat.defeat=0){
-    for (i=0;i<array_length(unit_struct);i++){
-        unit=unit_struct[i];
-        if (is_struct(unit)){
-            if (marine_dead[i]=0) and (obj_ncombat.player_max<obj_ncombat.enemy_max) and (ally[i]==false){
-                new_exp=0;
-                cur_exp=unit.experience;
-                if (cur_exp>=40) then new_exp+=choose(0,0,1);
-                if (cur_exp>=20) and (cur_exp<40) then new_exp+=choose(0,1);
-                if (cur_exp<20) then new_exp+=1;
-                
-                if (obj_ncombat.enemy=10) and (obj_ncombat.threat=7){
-                    if (cur_exp>=40) then new_exp+=choose(2,3,4);
-                    if (cur_exp>=20) and (cur_exp<40) then new_exp+=choose(4,6,8);
-                    if (cur_exp<20) then new_exp+=10;
-                }
-                if (new_exp>0){
-                    unit.add_exp(new_exp);
-                    obj_ncombat.total_battle_exp_gain+=new_exp;
-                }
-                if (unit.IsSpecialist("libs")) then unit.update_powers();
-                // Need some kind of report here
-            }
+if (obj_ncombat.defeat == 0) {
+    obj_ncombat.total_battle_exp_gain = obj_ncombat.threat * 50;
+    var _current_exp;
+    var _eligible_units = [];
+    var _exp_mod = 1;
+    var _unit_xp_data = [];
+    var _unit_recovery_score = obj_ncombat.unit_recovery_score;
+
+    for (var i = 0; i < array_length(unit_struct); i++) {
+        _unit = unit_struct[i];
+        if (is_struct(_unit) && ally[i] == false) {
+            if (marine_dead[i] == 1 && marine_type[i] != "" && _unit_recovery_score > 0) { // Apothecaries saving marines
+                obj_ncombat.unit_recovery_score -= 1;
+                _unit.update_health(irandom(20) - 10);
+                marine_dead[i] = false;
+                obj_ncombat.units_saved += 1;
+            } 
             
-            if (marine_type[i]!="") and (marine_dead[i]==1) and (ally[i]=false){
-                if (marine_type[i]=="Chapter Master"){
-                    if (obj_ncombat.unit_recovery_score>0){
-                        obj_ncombat.unit_recovery_score-=1;
-                        unit.update_health(irandom(20)-10);
-                        marine_dead[i]=0;
-                        obj_ncombat.units_saved+=1;                
-                    }
-                }else if (marine_type[i]!=""){
-                    if (obj_ncombat.unit_recovery_score>0){
-                        obj_ncombat.unit_recovery_score-=1;
-                        unit.update_health(irandom(20)-10);
-                        marine_dead[i]=0;
-                        obj_ncombat.units_saved+=1;
-                        // show_message(string(marine_type[i])+" is saved by an apothecary");
-                    }
-                }
+            if (!marine_dead[i]) { // EXP allocation
+                _current_exp = _unit.experience;
+
+                _exp_mod = 1 - (_current_exp / 200);
```

Comment:
> limit at a minimum of 3% otherwise exp growth will stop at 200exp?

Oh, yeah. Totally. Good catch.
This is what I was talking about when saying that the formula is junk.